packages/api/src/auth/auth.service.ts

```diff
@@ -1,153 +1,107 @@
+import { forwardRef, Inject, NotFoundException, UnauthorizedException } from '@nestjs/common';
 import jsonwebtoken from 'jsonwebtoken';
 import { UserEntity } from '../database/entities/user.entity';
-import { CreateUserDto } from '../users/dto/create-user.dto';
-// eslint-disable-next-line import/no-cycle
 import { UsersService } from '../users/users.service';
-import { Role } from './role.enum';
+import { TokenConfig } from '../utils/constants';
+import { CreateUserDto } from '../users/dto/create-user.dto';
 
-export interface TokenPayload {
-  email_verified: boolean;
-  auth_time: number;
-  exp: number;
-  email: string;
-  group?: Array<Role>;
+export interface Tokens {
+  accessToken: string;
+  refreshToken: string;
+  logoutToken: string;
 }
-
-interface CognitoTokenContent {
-  header: {
-    alg: string;
-    kid: string;
-  };
-  payload: TokenPayload;
-  signature: string;
+export interface TokenPayload {
+  sub: string;
+  roles: Array<string>;
+  keepMeLogin: boolean;
 }
 
-@Injectable({ scope: Scope.REQUEST })
 export class AuthService {
-  private tokenContent: TokenPayload | undefined;
-
   constructor(
     @Inject(forwardRef(() => UsersService))
     private readonly usersService: UsersService,
-    private readonly jwtService: JwtService,
-    @Inject(REQUEST)
-    private readonly request: Request,
   ) {}
 
-  public async isRequestAuthenticated(requiredRole?: Role): Promise<boolean> {
-    try {
-      const token: string = this.extractTokenFromHeaders();
+  public generateTokens(user: UserEntity, keepMeLogin: boolean): Tokens {
+    const payload: TokenPayload = { sub: user.id, roles: user.roles, keepMeLogin };
+    const tokens: Tokens = {
+      accessToken: jsonwebtoken.sign(payload, 'secret', {
+        expiresIn: TokenConfig.accessToken.expiresIn,
+      }),
+      logoutToken: keepMeLogin
+        ? ''
+        : jsonwebtoken.sign(payload, 'secret', {
+            expiresIn: TokenConfig.logoutToken.expiresIn,
+          }),
+      refreshToken: jsonwebtoken.sign(payload, 'secret', {
+        expiresIn: keepMeLogin
+          ? TokenConfig.refreshToken.keepMe.expiresIn
+          : TokenConfig.refreshToken.withOutKeepMe.expiresIn,
+      }),
+    };
 
-      await this.decodeAndVerifyToken(token);
+    // todo store refresh token in db
 
-      if (requiredRole) {
-        this.verifyRole(requiredRole);
-      }
-
-      return true;
-    } catch (error) {
-      console.error('Error: ', error);
-      return false;
-    }
+    return tokens;
   }
 
-  async validateUser(email: string, pass: string): Promise<any> {
-    const user = await this.usersService.findOneByEmail(email);
-    if (user && user.password === pass) {
-      const { password, ...result } = user;
-      return result;
+  public validateTokens(tokens: Tokens): TokenPayload {
+    const { accessToken, logoutToken } = tokens;
+    try {
+      const accessPayload = jsonwebtoken.verify(accessToken, 'secret') as TokenPayload;
```

Comment:
Please use `:` types notation instead of `as` to keep consistency in the whole codebase. Syntax with `as` is equivalent to `:` and works the same but it was introduced mainly for `.tsx` file extensions.